geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/ExportDataCommand.java

```diff
@@ -87,4 +80,45 @@ public Result exportData(
     }
     return result;
   }
+
+  private Result getFunctionResult(ResultCollector<?, ?> rc) {
+    Result result;
+    List<Object> results = (List<Object>) rc.getResult();
+    if (results != null) {
+      Object resultObj = results.get(0);
+      if (resultObj instanceof String) {
+        result = ResultBuilder.createInfoResult((String) resultObj);
+      } else if (resultObj instanceof Exception) {
+        result = ResultBuilder.createGemFireErrorResult(((Exception) resultObj).getMessage());
+      } else {
+        result = ResultBuilder.createGemFireErrorResult(
+            CliStrings.format(CliStrings.COMMAND_FAILURE_MESSAGE, CliStrings.EXPORT_DATA));
+      }
+    } else {
+      result = ResultBuilder.createGemFireErrorResult(
+          CliStrings.format(CliStrings.COMMAND_FAILURE_MESSAGE, CliStrings.EXPORT_DATA));
+    }
+    return result;
+  }
+
+  private String defaultFileName(String dirPath, String regionName) {
+    return new File(dirPath, regionName + RegionSnapshotService.SNAPSHOT_FILE_EXTENSION)
+        .getAbsolutePath();
+  }
+
+  private Optional<Result> validatePath(String filePath, String dirPath, boolean parallel) {
```

Comment:
this seems to be duplicated between import/export, any way we can use the same code base?

Reply:
I thought about having them both extend the same abstract base class: that would remove the duplicate code. The method is not dependant on state of the object, so it could be extracted into a static helper method (suggestions on where would be best to put it if this route is desirable?).

Reply:
Our StartLocatorCommand and StartServerCommand had a similar problem, we eventually used a StartMemberUtil class to hold the common code. Duplicate or static, had to choose a less evil.